src/ncp/ncp_spinel.cpp

```diff
@@ -354,20 +338,42 @@ void NcpSpinel::HandleValueIs(spinel_prop_key_t aKey, const uint8_t *aBuffer, ui
     return;
 }
 
-void NcpSpinel::HandleResponseForCommand(spinel_tid_t aTid, otError aError)
+otbrError NcpSpinel::HandleResponseForPropSet(spinel_tid_t      aTid,
+                                              spinel_prop_key_t aKey,
+                                              const uint8_t    *aData,
+                                              uint16_t          aLength)
 {
-    switch (mCmdTable[aTid])
+    OTBR_UNUSED_VARIABLE(aData);
+    OTBR_UNUSED_VARIABLE(aLength);
+
+    otbrError error = OTBR_ERROR_NONE;
+
+    switch (mWaitingKeyTable[aTid])
     {
-    case SPINEL_CMD_NET_CLEAR:
+    case SPINEL_PROP_THREAD_ACTIVE_DATASET:
+    {
+        VerifyOrExit(aKey == SPINEL_PROP_THREAD_ACTIVE_DATASET, error = OTBR_ERROR_INVALID_STATE);
+        CallAndClear(mDatasetSetActiveTask, OT_ERROR_NONE);
+        break;
+    }
+    case SPINEL_PROP_NET_IF_UP:
+    {
+        VerifyOrExit(aKey == SPINEL_PROP_NET_IF_UP, error = OTBR_ERROR_INVALID_STATE);
+        CallAndClear(mIp6SetEnabledTask, OT_ERROR_NONE);
+        break;
+    }
+    case SPINEL_PROP_NET_STACK_UP:
     {
-        CallAndClear(mThreadErasePersistentInfoTask, aError);
+        VerifyOrExit(aKey == SPINEL_PROP_NET_STACK_UP, error = OTBR_ERROR_INVALID_STATE);
+        CallAndClear(mThreadSetEnabledTask, OT_ERROR_NONE);
         break;
     }
     default:
         break;
```

Comment:
Does it make sense in `default` case to do `VerifyOrExit(aKey == mWaitingKeyTable[aTid], error = ...)`?

(since we remove a similar check in `HandleResponse()` in `VerifyOrExit(cmd == SPINEL_CMD_PROP_VALUE_IS && key == mWaitingKeyTable[aTid], error = OTBR_ERROR_INVALID_STATE);` ?

Reply:
Make sense! Updated.